### PR TITLE
Refactor course_catalog course parsing

### LINKED ISSUE #1838: refactor course catalog helpers
In our haste to merge the course catalog, we didn't address all the issues on the pull request. 

See, in particular, https://github.com/mitodl/open-discussions/pull/1753#discussion_r253616222

### LINKED ISSUE #1849: Split up digest_ocw_course into separate methods
Simplify and split up digest_ocw_course into separate methods because pylint is complaining about `too-many-locals` and `too-many-branches`

#### Acceptance Criteria:

- [ ] Pylint errors `too-many-locals` and `too-many-branches` no longer need to be disabled for this method.

### PR DESCRIPTION
**Pull request by sesterki**
#### What are the relevant tickets?
closes #1838 
closes #1849

#### What's this PR do?
Refactor course sync to utilize custom serializers, to remove pylint exceptions, and to remove duplicated code.

#### How should this be manually tested?
Open the shell and manually run the various course sync functions (edx/mitx, ocw, bootcamps). Make sure that all of the data was correctly populated.

### CODE REVIEW

File: course_catalog/serializers.py [hunk 1]
```diff
@@ -56,7 +110,138 @@ class Meta:
         extra_kwargs = {"raw_json": {"write_only": True}}
 
 
-class BootcampSerializer(serializers.ModelSerializer):
+class CourseSerializerTemporary(CourseRelatedFieldSerializer):
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/serializers.py]: Done

File: course_catalog/serializers.py [hunk 2]
```diff
@@ -56,7 +110,138 @@ class Meta:
...
+            "image_src": data.get("image_src"),
+            "image_description": data.get("image_description"),
+            "last_modified": data.get("last_modified"),
+            "published": data.get("is_published"),
+            "raw_json": data.get("raw_json"),
+            "url": get_course_url(data.get("uid"), data, PlatformType.ocw.value),
+            "availability": AvailabilityType.current.value,
+        }
+        if "PROD/RES" in data.get("course_prefix"):
+            course_fields["learning_resource_type"] = ResourceType.ocw_resource.value
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/serializers.py]: Done

File: course_catalog/serializers.py [hunk 3]
```diff
@@ -41,7 +48,54 @@ class Meta:
         fields = "__all__"
 
 
-class CourseSerializer(serializers.ModelSerializer):
+class CourseRelatedFieldSerializer(serializers.ModelSerializer):
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/serializers.py]: Done

File: course_catalog/serializer_helpers_test.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+"""
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/serializer_helpers_test.py]: Done

File: course_catalog/serializer_helpers.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
+"""
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/serializer_helpers.py]: Done

File: course_catalog/serializers.py [hunk 6]
```diff
@@ -41,7 +48,54 @@ class Meta:
...
+        self.handle_many_to_many(course)
+        return course
+
+    def update(self, instance, validated_data):
+        """
+        Custom update since serializers don't do writable nested serializers by default
+        """
+        course = super().update(instance, validated_data)
+        self.handle_many_to_many(course)
+        return course
```
[the remark this answers is not in the document]
Comment (sesterki) [reply course_catalog/serializers.py]: Done

**Comment by sesterki** [comment]
I was just about to ask about that. Thanks.

**Comment by aliceriot** [comment]
@sesterki can you squash your PR down do a single commit before merging next time? that's or normal practice, you can do it right in the github UI if you use that for merging PRs.

**Pull request merged**